### ISSUE TITLE
Failure of auto-batching

### ISSUE THREAD

**Issue by slaweks17** [clab/dynet#1178]
Hi,
My program works OK without auto batching. When I try to use it, after a few batches are processed, I am getting ""vector subscript out of range" from std::vector, caused by executing the second line from the following:
Expression batchLoss_ex=sum(batchLoss_exVect);
float loss = as_scalar(cg.forward(batchLoss_ex));

batchLoss_exVect is defined as vector<Expression>;

The stack trace suggests it happens while trying to add sum function to the graph. In 
inline VariableIndex ComputationGraph::add_function_node(Node *node), line
  node->device = nodes[node->args[0]]->device;
Apparently node->args[0] is too big (something like 1939).

Anyone has any ideas? It is Windows, VS 2015, a 2.02 version of Dynet I believe (downloaded a few days ago), compiled with also recent MKL.

Regards,
Slawek

**Comment by neubig** [comment]
This sounds like a bug, but it would very much help us debug if you could create a minimal working example (e.g. a single C++ program that can reproduce the bug on its own). Could you see if you can isolate a minimal piece of code that makes it possible to reproduce the problem?

**Comment by slaweks17** [comment]
Hi,
Thank you. I will, although it will take me a bit of time.

Regards,
Slawek

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by neubig** [comment]
Closing for lack of activity, but feel free to open if you can provide an example
